source/patterns/@aws-solutions-constructs/aws-fargate-kinesisfirehose/lib/index.ts

```diff
@@ -0,0 +1,183 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import * as ec2 from "aws-cdk-lib/aws-ec2";
+import * as kinesisfirehose from "aws-cdk-lib/aws-kinesisfirehose";
+// Note: To ensure CDKv2 compatibility, keep the import statement for Construct separate
+import { Construct } from "constructs";
+import * as defaults from "@aws-solutions-constructs/core";
+import * as ecs from "aws-cdk-lib/aws-ecs";
+import * as iam from "aws-cdk-lib/aws-iam";
+
+export interface FargateToKinesisFirehoseProps {
+  /**
+   * Optional custom properties for a VPC the construct will create. This VPC will
+   * be used by the new Fargate service the construct creates (that's
+   * why targetGroupProps can't include a VPC). Providing
+   * both this and existingVpc is an error. A Kinesis Firehose Interface
+   * endpoint will be included in this VPC.
+   *
+   * @default - none
+   */
+  readonly vpcProps?: ec2.VpcProps;
+  /**
+   * An existing VPC in which to deploy the construct. Providing both this and
+   * vpcProps is an error. If the client provides an existing Fargate service,
+   * this value must be the VPC where the service is running. A Kinesis Interface
+   * endpoint will be added to this VPC.
+   *
+   * @default - none
+   */
+  readonly existingVpc?: ec2.IVpc;
+  /**
+   * True if the VPC provisioned by this construct should contain Public/Private Subnets,
+   * otherwise False for the VPC to contain Isolated Subnets only. Note this property is
+   * ignored if an existing VPC is specified in the existingVpc property. If you are getting
+   * a container from a public repo, this must be true so the repo can be accessed from the
+   * network.
+   *
+   * @default - none
```

Comment:
Nit based on the last fargate-kinesisstreams PR - remove the `@default - none` since this ins't an optional property.